HospitalSystem.Core/Appointments/AppointmentService.cs

```diff
@@ -0,0 +1,399 @@
+
+            if (!(interval.Starts.TimeOfDay<=iterationDate.TimeOfDay && iterationDate.TimeOfDay<interval.Ends.TimeOfDay))
+            {
+                iterationDate = iterationDate.AddMinutes(15);
+                continue;
+            }
+
+            foreach (Doctor doctor in _doctorService.GetManyBySpecialty(speciality))
+            {
+                Checkup newCheckup = new Checkup(
+                    iterationDate,
+                    new MongoDB.Driver.MongoDBRef("patients", user.Person.Id),
+                    new MongoDB.Driver.MongoDBRef("doctors", doctor.Id),
+                    "no anamnesis");
+                if (!IsDoctorAvailable(newCheckup.DateRange, doctor))
+                {
+                    continue;
+                }
+                else
+                {
+                    if (checkups.Count >= numberOfCheckups)
+                    {
+                        break;
+                    }
+
+                    checkups.Add(newCheckup);
+                }
+            }
+            iterationDate = iterationDate.AddMinutes(15);
+        }
+        return checkups;
+    }
+
+    public List<Checkup> GetFirstFewFreeCheckups(Doctor doctor, int numberOfCheckups, User user)
+    {
+        List<Checkup> checkups = new List<Checkup>();
+        DateTime iterationDate = RoundUp(DateTime.Now, TimeSpan.FromMinutes(15));
+
+        while (checkups.Count < numberOfCheckups)
+        {
+            if (iterationDate.TimeOfDay >=Globals.ClosingTime.TimeOfDay)
+            {
+                iterationDate = new DateTime(iterationDate.Year, iterationDate.Month, iterationDate.Day,
+                    Globals.OpeningTime.Hour, Globals.OpeningTime.Minute, Globals.OpeningTime.Second);
+                iterationDate = iterationDate.AddDays(1);
+                continue;
+            }
+
+            if (iterationDate.TimeOfDay < Globals.OpeningTime.TimeOfDay)
+            {
+                iterationDate = new DateTime(iterationDate.Year, iterationDate.Month, iterationDate.Day,
+                    Globals.OpeningTime.Hour, Globals.OpeningTime.Minute, Globals.OpeningTime.Second);
+                continue;
+            }
+            
+            Checkup newCheckup = new Checkup(
+                iterationDate,
+                new MongoDB.Driver.MongoDBRef("patients", user.Person.Id),
+                new MongoDB.Driver.MongoDBRef("doctors", doctor.Id),
+                "no anamnesis");
+            if (!IsDoctorAvailable(newCheckup.DateRange, doctor))
+            {
+                iterationDate = iterationDate.AddMinutes(15);
+                continue;
+            }
+            else
+            {
+                checkups.Add(newCheckup);
+                iterationDate = iterationDate.AddMinutes(15);
+            }
+        }
+        return checkups;
+    }
+
+    public DateTime RoundUp(DateTime dt, TimeSpan d)
+    {
+        return new DateTime((dt.Ticks + d.Ticks - 1) / d.Ticks * d.Ticks, dt.Kind);
+    }
```

Comment:
Wrong spot. Does not belong in service. move to some utils. Could do this on datetime